src/modules/media/elasticsearch/queryBuilder.ts

```diff
@@ -80,116 +85,112 @@ export class QueryBuilder {
+			};
+		}
+		return searchFilter.multiValue || searchFilter.value;
+	}
+
+	protected static getQueryType(searchFilter: SearchFilter, value: any): any {
+		const defaultQueryType = this.config.DEFAULT_QUERY_TYPE[searchFilter.field];
+		if (defaultQueryType === QueryType.TERMS && !_.isArray(value)) {
+			return QueryType.TERM;
+		}
+		return defaultQueryType;
+	}
+
+	protected static buildFilter(elasticKey: string, searchFilter: SearchFilter): any {
+		const occurrenceType = this.getOccurrenceType(searchFilter.operator);
+		const value = this.buildValue(searchFilter);
+		return {
+			occurrenceType,
 			query: {
-				[this.config.DEFAULT_QUERY_TYPE[readableKey]]: {
-					[elasticKey + this.suffix(readableKey)]: advancedQuery[key],
+				[this.getQueryType(searchFilter, value)]: {
+					[elasticKey + this.suffix(searchFilter.field)]: value,
 				},
 			},
-		}));
+		};
+	}
 
-		return result;
+	protected static buildFreeTextFilter(searchTemplate: any, searchFilter: SearchFilter): any {
+		// Replace {{query}} in the template with the escaped search terms
+		const textQueryFilterArray = _.cloneDeep(searchTemplate);
+		const escapedQueryString = searchFilter.value;
+		_.forEach(textQueryFilterArray, (matchObj) => {
+			_.set(matchObj, 'multi_match.query', escapedQueryString);
+		});
+
+		return textQueryFilterArray;
 	}
 
 	/**
 	 * Creates the filter portion of the elasticsearch query object
 	 * Containing the search terms and the checked filters
 	 * @param filters
 	 */
-	private static buildFilterObject(filters: Partial<SearchFilters> | undefined) {
+	private static buildFilterObject(filters: SearchFilter[] | undefined) {
 		if (!filters || _.isEmpty(filters)) {
 			// Return query object that will match all results
 			return { match_all: {} };
 		}
 
 		const filterObject: any = {};
-		const stringQuery = _.get(filters, 'query');
-		if (stringQuery) {
-			// Replace {{query}} in the template with the escaped search terms
-			const textQueryObjectArray = _.cloneDeep(searchQueryObjectTemplate);
-			const escapedQueryString = stringQuery;
-			_.forEach(textQueryObjectArray, (matchObj) => {
-				_.set(matchObj, 'multi_match.query', escapedQueryString);
-			});
-
-			_.set(filterObject, 'bool.must', textQueryObjectArray);
-
-			if (_.keys(filters).length === 1) {
-				// Only a string query is passed, no need to add further filters
-				return filterObject;
-			}
-		}
 
 		// Add additional filters to the query object
 		const filterArray: any[] = [];
 		_.set(filterObject, 'bool.filter', filterArray);
-		_.forEach(filters, (value: any, readableKey: keyof SearchFilters) => {
-			if (readableKey === 'query') {
-				return; // Query filter has already been handled, skip this foreach iteration
+		_.forEach(filters, (searchFilter: SearchFilter) => {
+			if (
+				searchFilter.field === SearchFilterField.QUERY ||
+				searchFilter.field === SearchFilterField.ADVANCED_QUERY
+			) {
+				if (!searchFilter.value) {
+					// empty value, ignore
```

Comment:
wouldn't it be better to throw an error here? Just to be strict and not hide any bugs in the client?